generators/rabbitmq/files.mjs

```diff
@@ -10,39 +11,118 @@ import { moveToJavaPackageSrcDir } from '../server/support/index.mjs';
  */
 
 export const rabbitFiles = {
-    config: [
-      {
-        path: `${SERVER_MAIN_SRC_DIR}package/`,
-        renameTo: moveToJavaPackageSrcDir,
-        templates: ['config/RabbitMQConfig.java'],
-      },
-    ],
-    domain: [
-      {
-        path: `${SERVER_MAIN_SRC_DIR}package/`,
-        renameTo: moveToJavaPackageSrcDir,
-        templates: ['domain/RabbitMessageModel.java'],
-      },
-    ],
-    listener: [
-      {
-        path: `${SERVER_MAIN_SRC_DIR}package/`,
-        renameTo: moveToJavaPackageSrcDir,
-        templates: ['listener/RabbitMQConsumer.java'],
-      },
-    ],
-    publisher: [
+  config: [
+    {
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['config/RabbitMQConfig.java'],
+    },
+  ],
+  domain: [
+    {
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['domain/RabbitMessageModel.java'],
+    },
+  ],
+  listener: [
+    {
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['listener/RabbitMQConsumer.java'],
+    },
+  ],
+  publisher: [
+    {
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['publisher/RabbitMQPublisher.java'],
+    },
+  ],
+};
+
+export default async function writeRabbitMQFilesTask({ application }) {
+  if (communications.length === 0) {
+    application.queueName = application.baseName;
+    application.uniqueName = null;
+    this.writeFiles({
+      sections: rabbitFiles,
+      context: application,
+    });
+  } else {
+    for (let i = 0; i < communications.length; i++) {
+      if (this.jhipsterConfig.baseName === communications[i].client) {
```

Comment:
listener has to be added even when the cummincations.length === 0 ryt ?
if (this.jhipsterConfig.baseName === communications[i].client || communications.length === 0) {
}